From d4479e36c2cda60dac4f7ea450c4055efd671f41 Mon Sep 17 00:00:00 2001
From: Robby Cai <R63905@freescale.com>
Date: Thu, 27 Mar 2014 18:55:21 +0800
Subject: [PATCH 0359/1074] ENGR00305598-4 ARM: dts: imx6sx: add new DTS file
 to enable LCDIF1

Due to the CSI and LCDIF1 shares the same pin MX6SX_PAD_LCD1_ENABLE.

Signed-off-by: Robby Cai <R63905@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 arch/arm/boot/dts/Makefile                     |    1 +
 arch/arm/boot/dts/imx6sx-19x19-arm2-lcdif1.dts |   25 ++++++++++++++++++++++++
 2 files changed, 26 insertions(+), 0 deletions(-)
 create mode 100644 arch/arm/boot/dts/imx6sx-19x19-arm2-lcdif1.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index c246bfe..327c64f 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -184,6 +184,7 @@ dtb-$(CONFIG_ARCH_MXC) += \
 	imx6sl-evk-pf200.dtb \
 	imx6sx-17x17-arm2.dtb \
 	imx6sx-19x19-arm2.dtb \
+	imx6sx-19x19-arm2-lcdif1.dtb \
 	imx6sx-sdb.dtb \
 	imx6sx-sdb-lcdif1.dtb \
 	vf610-cosmic.dtb \
diff --git a/arch/arm/boot/dts/imx6sx-19x19-arm2-lcdif1.dts b/arch/arm/boot/dts/imx6sx-19x19-arm2-lcdif1.dts
new file mode 100644
index 0000000..d715fd5
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sx-19x19-arm2-lcdif1.dts
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sx-19x19-arm2.dts"
+
+/ {
+	csi1_v4l2_cap {
+		status = "disabled";
+	};
+};
+
+&i2c2 {
+	ov564x: ov564x@3c {
+		status = "disabled";
+	};
+};
+
+&lcdif1 {
+	status = "okay";
+};
-- 
1.7.5.4

